ui/perfherder/alerts/AlertsView.jsx

```diff
@@ -133,9 +133,11 @@ class AlertsView extends React.Component {
 
     if (this.isListMode()) {
       if (doUpdateParams) {
-        this.props.validated.updateParams(
-          this.getParamsFromFilters(updatedFilters),
-        );
+        const updateParameters = this.getParamsFromFilters(updatedFilters);
+        if (!updatedFilters.hasOwnProperty('page')) {
```

Comment:
Why do you need to check if updatedFilters has the property page.
Otherwise it works well.

Reply:
Ensuring there is a default `page=1` should be in the `getParamsFromFilters` function: 

Instead of 
```
  getParamsFromFilters = filters => {
    const params = {};
    for (const [filterName, filterValue] of Object.entries(filters)) {
      Object.assign(params, this.mapParamFromFilter(filterName, filterValue));
    }
    return params;
  };
```

add the default `page` in the return like

```
  getParamsFromFilters = filters => {
    const params = {};
    for (const [filterName, filterValue] of Object.entries(filters)) {
      Object.assign(params, this.mapParamFromFilter(filterName, filterValue));
    }
    return {page: 1, ...params};
  };
```


In general, try to find the deepest function call that will fix the problem.  If you use an `if` block, you are probably solving the problem in the wrong place.

Reply:
The `getParamsFromFilters` is a bit more complicated than it need be. Inline `mapParamFromFilter` into this function, and simplify.  From  

```es6
  getParamsFromFilters = filters => {
    const params = {};
    for (const [filterName, filterValue] of Object.entries(filters)) {
      Object.assign(params, this.mapParamFromFilter(filterName, filterValue));
    }
    return params;
  };

  mapParamFromFilter = (filterName, filterValue) => {
    switch (filterName) {
      case 'framework':
        return { [filterName]: filterValue.id };
      case 'status':
        return { [filterName]: summaryStatusMap[filterValue] };
      case 'hideDownstream':
        return { hideDwnToInv: +filterValue };
      case 'hideImprovements':
      case 'hideAssignedToOthers':
        return { [filterName]: +filterValue };
      default:
        return { [filterName]: filterValue };
    }
  };
```


to 

```es6
  getParamsFromFilters = filters => {
    return {
      page: 1,  //default value
      ...Object.fromEntries(
        Object.entries(filters).map(([filterName, filterValue])=>{
          switch (filterName) {
            case 'framework':
              return [filterName, filterValue.id ];
            case 'status':
              return [filterName, summaryStatusMap[filterValue] ];
            case 'hideDownstream':
              return ["hideDwnToInv",  +filterValue ];
            case 'hideImprovements':
            case 'hideAssignedToOthers':
              return [filterName,  +filterValue ];
            default:
              return [filterName,  filterValue ];
          }
        })
      )
    }
  };
```